### DISCUSSION TITLE
Docker container for test deploys.

### DISCUSSION THREAD

**Discussion by danshumaker** [deployphp/deployer#2840] [General]
I'm trying to debug why the writable task doesn't do what I want. ( https://github.com/deployphp/deployer/blob/master/recipe/deploy/writable.php )

In testing deploys it is very tedious to use our actual servers (via gitlab) and am curious if anyone has setup a temporary docker container and provided deployerphp the host address of the docker container - and then deployed to that (as a test of production).

I'm currently using docksal for local development.  Docksal doesn't need to be deployed to since the files are synchronized via a mount.  But I guess I could repurpose the already built containers that Docksal has, so that I could test deployer with it.. But I'm not sure where to begin with that.  The first thing I think would be to use the same docker images but take out the mounts.  

**Has anyone tried deploying to local docker containers and can provide documentation or help on a quick setup?**

To make a long story short, when I call deploy:writable it looks like the writable_dirs array is empty and so it immediately returns without doing anything - but it's a challenge to debug this.  And it's really weird that they are emptied on line 41 ( https://github.com/deployphp/deployer/blob/master/recipe/deploy/writable.php#L41 ).

**Comment by pokgak** [comment]
Never tried it myself but here's how I imagined I would do this:
- make sure container runs SSH, nginx, php
- get ip of the container
- use that ip as the target host in deploy.php

Running multiple services in a single docker container is not how docker is meant to be used though. Something like Vagrant might be better if you're running the test locally but I'm not sure if gitlab ci allows you to run vagrant.

**Comment by antonmedv** [comment]
Take a look at our e2e tests. It’s one container deploying to another one.
